.github/workflows/firesim-run-tests.yml

```diff
@@ -164,16 +164,27 @@ jobs:
       - name: Run pytests
         run: .github/scripts/run-manager-pytests.py
 
-  run-python-typecheck:
-    name: run-python-typecheck
+  run-manager-python-typecheck:
+    name: run-manager-python-typecheck
     needs: [setup-manager]
     runs-on: aws-${{ github.run_id }}
     env:
       TERM: xterm-256-color
     steps:
       - uses: actions/checkout@v3
-      - name: Run mypy type checker
-        run: .github/scripts/run-mypy-typechecker.py
+      - name: Run manager Python type checker
+        run: .github/scripts/run-manager-python-typechecking.py
+
+  run-ci-python-typecheck:
```

Comment:
Just a thought, but what if we made the ci typecheck the first script run in the CI pipeline? This might be a bit too strict, but the idea would be to check earlier on and maybe make this a dependency that forces us to fix typing before testing CI. Just a suggestion, this might be to anal.

Reply:
I agree this should be done. However, I think the best way to do this is to setup a way to initialize the FireSim repo w/ GH-A managed runners and move this job there so that even spawning the manager is blocked by a typing check. I don't have time to do this, so I'll leave this to you or just future work.

Reply:
Yeah that's what I was suggesting, I'll put it in my backlog